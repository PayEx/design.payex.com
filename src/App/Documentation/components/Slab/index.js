import React, { Component } from "react";

import { ComponentPreview, DocContainer, Property } from "#";
import { validation } from "$/px-script/main/";

const DefaultSlab = () => (
    <>
        <h2 id="default-slab">Default slab</h2>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab">
                <p>Do you want to sign up for the slab mailing list?</p>
                <form action="#" noValidate data-validate>
                    <div className="form-group">
                        <label htmlFor="validation-email-2">Email</label>
                        <div className="input-group">
                            <span className="input-group-addon"><i className="material-icons">email</i></span>
                            <input type="email" className="form-control" id="validation-email-2" placeholder="bob.corlsan@example.com" required />
                        </div>
                        <div className="help-block" data-success="Right!" data-error="Wrong!">This one might be a little tricky</div>
                    </div>
                    <button className="btn btn-executive" type="submit">Submit</button>
                </form>
            </div>
        </ComponentPreview>
    </>
);

const SlabElevated = () => (
    <>
        <h2 id="elevated-slab">Elevated slab</h2>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab slab-elevated">
                Look ma, im elevated!
            </div>
        </ComponentPreview>
    </>
);

const SlabWell = () => (
    <>
        <h2 id="slab-well">Slab well</h2>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab slab-well">
                Look ma, im in a well!
            </div>
        </ComponentPreview>
    </>
);

const SlabArrow = () => (
    <>
        <h2 id="slab-arrow">Slab arrow</h2>
        <p>
            <Property val=".slab-arrow" /> inherits the color from the slab it is appended to, and is given a border
            if used with slab-white.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab slab-white slab-arrow">
                <p>
                    This slab has some connection to the content above.
                </p>
            </div>
        </ComponentPreview>
    </>
);

const SlabColors = () => (
    <>
        <h2 id="slab-colors">Slab colors</h2>
        <p>
            Both white and apricot variants of slab exist.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab slab-white">
                <h4>Here is some text and then a striped item list</h4>
                <p>
                    Some text with a small listing of items underneath.
                    <ul className="list item-list">
                        <li>Item 1</li>
                        <li>Item 2</li>
                        <li>Item 3</li>
                    </ul>
                </p>
            </div>

            <div className="slab slab-apricot">
                <h4>Here is some text and then a striped item list</h4>
                <p>
                    Some text with a small listing of items underneath.
                    <ul className="list item-list">
                        <li>Item 1</li>
                        <li>Item 2</li>
                        <li>Item 3</li>
                    </ul>
                </p>
            </div>
        </ComponentPreview>
    </>
);

const SlabMuted = () => (
    <>
        <h2 id="slab-muted">Slab muted</h2>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab slab-muted">
                <h4>This text will be gray.</h4>{"\n"}
                <span>And this text will be gray.</span>{"\n"}
                Even this text will be gray!
            </div>
        </ComponentPreview>
    </>
);

const SlabSizes = () => (
    <>
        <h2 id="slab-sizes">Slab sizes</h2>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab slab-sm">
                This is very snug.
            </div>
        </ComponentPreview>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab slab-lg">
                This is so spacious I can hardly believe it!
            </div>
        </ComponentPreview>
    </>
);

const SlabCombinations = () => (
    <>
        <h2 id="slab-combinations">Slab combinations</h2>
        <p>
            You can mix and match slab variations as you see fit.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab slab-white slab-muted slab-elevated">
                This is an elevated white slab with muted text.
            </div>
        </ComponentPreview>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab slab-well slab-lg">
                This is a big well!
            </div>
        </ComponentPreview>
    </>
);

class Slab extends Component {
    componentDidMount () {
        validation.init();
    }

    render () {
        return (
            <DocContainer docToc>
                <p className="lead">
                    Slabs are container components, therefore it works with whatever you want to put in it.
                </p>
                <DefaultSlab />
                <SlabElevated />
                <SlabWell />
                <SlabArrow />
                <SlabColors />
                <SlabMuted />
                <SlabSizes />
                <SlabCombinations />
            </DocContainer>
        );
    }
}

export default Slab;

/* For testing */
export { DefaultSlab, SlabElevated, SlabWell, SlabArrow, SlabColors, SlabSizes, SlabMuted, SlabCombinations };
