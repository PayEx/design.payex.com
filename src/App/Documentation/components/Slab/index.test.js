import React from "react";
import { shallow } from "enzyme";

import Slab, { DefaultSlab, SlabElevated, SlabWell, SlabColors, SlabSizes, SlabMuted, SlabCombinations } from "./index";

describe("Components: Slab", () => {
    it("is defined", () => {
        expect(Slab).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Slab />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("DefaultSlab", () => {
        it("is defined", () => {
            expect(DefaultSlab).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<DefaultSlab />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("SlabElevated", () => {
        it("is defined", () => {
            expect(SlabElevated).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<SlabElevated />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("SlabWell", () => {
        it("is defined", () => {
            expect(SlabWell).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<SlabWell />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("SlabColors", () => {
        it("is defined", () => {
            expect(SlabColors).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<SlabColors />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("SlabMuted", () => {
        it("is defined", () => {
            expect(SlabMuted).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<SlabMuted />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("SlabSizes", () => {
        it("is defined", () => {
            expect(SlabSizes).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<SlabSizes />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("SlabCombinations", () => {
        it("is defined", () => {
            expect(SlabCombinations).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<SlabCombinations />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
