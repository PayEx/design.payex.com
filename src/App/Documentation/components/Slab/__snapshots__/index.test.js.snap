// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Slab DefaultSlab renders 1`] = `
<Fragment>
  <h2
    id="default-slab"
  >
    Default slab
  </h2>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="slab"
    >
      <p>
        Do you want to sign up for the slab mailing list?
      </p>
      <form
        action="#"
        data-validate={true}
        noValidate={true}
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="validation-email-2"
          >
            Email
          </label>
          <div
            className="input-group"
          >
            <span
              className="input-group-addon"
            >
              <i
                className="material-icons"
              >
                email
              </i>
            </span>
            <input
              className="form-control"
              id="validation-email-2"
              placeholder="bob.corlsan@example.com"
              required={true}
              type="email"
            />
          </div>
          <div
            className="help-block"
            data-error="Wrong!"
            data-success="Right!"
          >
            This one might be a little tricky
          </div>
        </div>
        <button
          className="btn btn-executive"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Slab SlabColors renders 1`] = `
<Fragment>
  <h2
    id="slab-colors"
  >
    Slab colors
  </h2>
  <p>
    Both white and apricot variants of slab exist.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="slab slab-white"
    >
      <h4>
        Here is some text and then a striped item list
      </h4>
      <p>
        Some text with a small listing of items underneath.
        <ul
          className="list item-list"
        >
          <li>
            Item 1
          </li>
          <li>
            Item 2
          </li>
          <li>
            Item 3
          </li>
        </ul>
      </p>
    </div>
    <div
      className="slab slab-apricot"
    >
      <h4>
        Here is some text and then a striped item list
      </h4>
      <p>
        Some text with a small listing of items underneath.
        <ul
          className="list item-list"
        >
          <li>
            Item 1
          </li>
          <li>
            Item 2
          </li>
          <li>
            Item 3
          </li>
        </ul>
      </p>
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Slab SlabCombinations renders 1`] = `
<Fragment>
  <h2
    id="slab-combinations"
  >
    Slab combinations
  </h2>
  <p>
    You can mix and match slab variations as you see fit.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="slab slab-white slab-muted slab-elevated"
    >
      This is an elevated white slab with muted text.
    </div>
  </ComponentPreview>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="slab slab-well slab-lg"
    >
      This is a big well!
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Slab SlabElevated renders 1`] = `
<Fragment>
  <h2
    id="elevated-slab"
  >
    Elevated slab
  </h2>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="slab slab-elevated"
    >
      Look ma, im elevated!
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Slab SlabMuted renders 1`] = `
<Fragment>
  <h2
    id="slab-muted"
  >
    Slab muted
  </h2>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="slab slab-muted"
    >
      <h4>
        This text will be gray.
      </h4>
      

      <span>
        And this text will be gray.
      </span>
      

      Even this text will be gray!
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Slab SlabSizes renders 1`] = `
<Fragment>
  <h2
    id="slab-sizes"
  >
    Slab sizes
  </h2>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="slab slab-sm"
    >
      This is very snug.
    </div>
  </ComponentPreview>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="slab slab-lg"
    >
      This is so spacious I can hardly believe it!
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Slab SlabWell renders 1`] = `
<Fragment>
  <h2
    id="slab-well"
  >
    Slab well
  </h2>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="slab slab-well"
    >
      Look ma, im in a well!
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Slab renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Slabs are container components, therefore it works with whatever you want to put in it.
  </p>
  <DefaultSlab />
  <SlabElevated />
  <SlabWell />
  <SlabArrow />
  <SlabColors />
  <SlabMuted />
  <SlabSizes />
  <SlabCombinations />
</DocContainer>
`;
